File: chapter2/insertion_sort.py
from random import shuffle
A = [i for i in range(10)]
shuffle(A) # shuffle() function has no return value, so A=shuffle(A) will cause an error.
print "sorting_array\n%s\n"%A
for j in range(1,10):
    key = A[j]
    i = j - 1
    while i>=0 and A[i]>key:
        A[i+1] = A[i]
        i -= 1
    A[i+1] = key
print "sorted_array:\n%s\n"%A

File: chapter2/insertion_sort.rb
A = (0..9).to_a.shuffle
puts "sorting_array:\n%s\n"%A.to_s
for j in 1..9
  key = A[j]
  i = j - 1
  while i >= 0 && A[i] > key
    A[i+1] = A[i]
    i -= 1
  end
  A[i+1] = key
end
puts "sorted_array:\n%s\n"%A.to_s
